### PR TITLE
Others/dutch auction poc

### PR DESCRIPTION
**Pull request by ernestpascual**
## Description of Changes
Added a dutch auction POC in BSC testnet for internal use. They use contracts from MISO.

[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDF-232)

### Developer Checklist:

* [x] I have followed the guidelines in our Contributing document
* [x] This PR has a corresponding JIRA ticket
* [x] My branch conforms with our naming convention i.e. `feature/HDF-XXX-description`
* [x] All files have appropriate file headers and documentation
* [ ] I have written new tests for your core changes, as applicable
* [ ] I have successfully ran tests locally

### Reviewers Checklist:
* [ ] Code is readable and understandable; any unclear parts have explanations 
* [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
* [ ] I have successfully ran tests locally

### CODE REVIEW

File: public/locales/en.json [hunk 1]
```diff
@@ -92,6 +92,8 @@
   "close": "Close",
   "farm": "Farm",
   "stake": "Stake",
+  "mint": "Mint",
+  "tokensale": "Token sale",
```
Comment (tracyarciaga) [inline public/locales/en.json]: Please capitalize "Token Sale" @ernestpascual

**Review by tracyarciaga — COMMENTED** [review]
Tested and token sale flow is working as expected. Few questions:
1. Can you double-check if the value of the deposit textbox does not reset after a successful tx in your end? I just encountered a weird behavior, the value of the deposit textbox was not reset when I tried it first time.
2. Would it be better if we display how many HALO tokens the user bought?

**Comment by ernestpascual** [comment]
@tracyarciaga added total claimable halo and average price. the price and data in the card updates every 10 seconds as well. 

cc: @ctverceles

File: src/halo-hooks/useMisoDutchAuction.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    const tokensClaimable = await dutchAuctionContract?.tokensClaimable(account)
+
+    setEndtime(auctionEndTime.toString())
+    setCurrentTokenPrice(formatNumber(+formatEther(currentTokenPrice)))
+    setMarketParticipationAgreement(marketParticipationAgreement)
+    setUserTokensClaimable(formatNumber(+formatEther(tokensClaimable)))
+    setAveragePrice(formatNumber(+formatEther(averagePrice)))
+  }, [dutchAuctionContract, account])
+
+  const fetchAllowance = useCallback(async () => {
```
Comment (schystz) [inline src/halo-hooks/useMisoDutchAuction.ts:40]: @ernestpascual 1 optional improvement here, is to use `useApproveCallback` for checking current allowance and approving the amount.

https://github.com/HaloDAO/halodao-interface/blob/ed91ba6256963b98913e02b957b63c363340138c/src/components/Farm/FarmPoolCard.tsx#L416

File: src/pages/Tokensale/TokenSalePanel.tsx [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+                  className="token-amount-input"
+                  value={depositValue}
+                  onUserInput={val => {
+                    onUserDepositInput(val)
+                  }}
+                />
+                {account && label !== 'To' && (
+                  <StyledBalanceMax
+                    style={{
+                      border: '1px solid #471BB2',
```
Comment (schystz) [inline src/pages/Tokensale/TokenSalePanel.tsx:253]: @ernestpascual been seeing stuff like this where the style is defined inline the JSX, I would avoid it and use StyledComponents instead, BUT it should be fine for POC stuff.

**Review by schystz — APPROVED** [review]
Some optional improvements but other than that works fine and code looks good to me!

File: src/halo-hooks/useMisoDutchAuction.ts (hunk 2 above)
Comment (ernestpascual) [reply src/halo-hooks/useMisoDutchAuction.ts:40]: oh cool this is useful!

**Review by tracyarciaga — APPROVED** [review]

**Pull request merged**